src/Utilities/Compiler/Extensions/IDictionaryExtensions.cs

```diff
@@ -28,5 +29,9 @@ internal static class IDictionaryExtensions
                 dictionary.Add(item);
             }
         }
+
+        public static bool IsEqualTo<TKey, TValue>(this IReadOnlyDictionary<TKey, TValue> dictionary, IReadOnlyDictionary<TKey, TValue> other)
```

Comment:
I did something generic for the value so I have lost the ability to pass the string comparison. Is it fine like this or do you prefer an extra comparer argument?

Reply:
I think it is fine to add it when required.